source/grid/grid_tools.cc

```diff
@@ -3665,6 +3665,11 @@ namespace GridTools
         AssertThrowMPI(ierr);
       }
 
+    // At this point, wait for all of the isend operations to finish:
```

Comment:
You could move this Waitall to the end of the function, which potentially could improve performance. Or am I wrong?

Reply:
I did think of this, but couldn't quite convince myself that that was right -- and then decided that for purposes of a fix for an actual problem, this is good enough. The function is improveworthy anyway for a variety of reasons, and that's on my list for post-release. I think as a bug fix, this is all I propose.